app/shop.py

```diff
@@ -0,0 +1,39 @@
+import datetime
+
+
+class Shop:
+    def __init__(self, name: str, location: list[int], products: dict) -> None:
```

Comment:
The type hint for `location` should be `tuple[int, int]` instead of `list[int]` if the location is meant to be a pair of coordinates. If it's a list of coordinates, then it's fine.